code/modules/scrap/oldificator.dm

```diff
@@ -0,0 +1,250 @@
+/obj/proc/make_old()
+	color = pick("#996633", "#663300", "#666666")
+	light_color = color
+	name = pick("old ", "expired ", "dirty ") + initial(name)
+	desc += pick(" Warranty has expired.", " The inscriptions on this thing were erased by time.", " Looks completely wasted.")
+
+	germ_level = pick(80,110,160)
+	if(prob(40))
+		if(prob(70))
+			light_power = light_power / pick(1.5, 2, 2.5)
+		if(prob(70))
+			light_range = light_range / pick(1.5, 2, 2.5)
+		if(prob(15))
+			light_range = 0
+			light_power = 0
+
+	for(var/obj/item/sub_item in contents)
+		sub_item.make_old()
+
+	/*
+	If anyone ever wishes to make a system
+	of critically broken items, and it's influence on them.
+	Feel free to.
+
+	if(prob(50))
+		crit_fail = 1
+	*/
+
+	update_icon()
+
+/obj/item/make_old()
+	..()
+	if(prob(75))
+		origin_tech = null
+	siemens_coefficient += 0.3
+
+/obj/item/weapon/tool/make_old()
+	..()
+	unreliability += rand(1, 5)
+
+/obj/item/weapon/storage/make_old()
+	var/del_count = rand(0,contents.len)
+	for(var/i = 1 to del_count)
+		var/removed_item = pick(contents)
+		contents -= removed_item
+		qdel(removed_item)
+
+	if(prob(75))
+		storage_slots = max(contents.len, max(0, storage_slots - pick(2, 2, 2, 3, 3, 4)))
+	if(prob(75))
+		max_storage_space = max_storage_space / 2
+
+	..()
+
+/obj/item/weapon/reagent_containers/make_old()
+	for(var/datum/reagent/R in reagents.reagent_list)
+		R.volume = rand(0, R.volume)
+	reagents.add_reagent("toxin", rand(0,10))
+	..()
+
+/obj/item/ammo_magazine/make_old()
+	var/del_count = rand(0,contents.len)
+	for(var/i = 1 to del_count)
+		var/removed_item = pick(stored_ammo)
+		stored_ammo -= removed_item
+		qdel(removed_item)
+	..()
+
+/obj/item/weapon/cell/make_old()
+	charge = max(0, rand(0, maxcharge * 2) - maxcharge)
+	if(prob(40))
+		rigged = TRUE
+		if(prob(80))
+			charge = maxcharge  //make it BOOM hard
+	..()
+
+/obj/item/weapon/stock_parts/make_old()
+	var/degrade = pick(0,1,1,1,2)
+	rating = max(rating - degrade, 1)
+	..()
+
+/obj/item/stack/material/make_old()
+	return
+
+/obj/item/stack/rods/make_old()
+	return
+
+/obj/item/weapon/tank/make_old()
+	var/new_vol = pick(0.2,0.4,0,6,0,8)
+	air_contents.gas["oxygen"] *= new_vol
+	air_contents.gas["carbon_dioxide"] *= new_vol
+	air_contents.gas["nitrogen"] *= new_vol
+	air_contents.gas["sleeping_agent"] *= new_vol
+	air_contents.temperature = 293
+	volume *= new_vol
+	air_contents.update_values()
+	..()
+
+/obj/item/weapon/circuitboard/make_old()
+	if(prob(75))
+		build_path = pick(/obj/machinery/washing_machine, /obj/machinery/broken, /obj/machinery/shower)
+	..()
+
+/obj/item/weapon/aiModule/make_old()
+	if(prob(75) && !istype(src, /obj/item/weapon/aiModule/broken))
+		var/obj/item/weapon/aiModule/brokenmodule = new /obj/item/weapon/aiModule/broken
+		brokenmodule.name = src.name
+		brokenmodule.desc = src.desc
+		brokenmodule.make_old()
+		qdel(src)
+	..()
+
+/obj/item/clothing/suit/space/make_old()
+	if(prob(75))
+		var/datum/breach/B = new()
```

Comment:
don't do this, there are functions to handle it. Look at random/voidsuit, i think i used it in there for spawning pre-damaged voidsuits